tutorials/networking/high_level_multiplayer.rst

```diff
@@ -371,9 +372,151 @@ have loaded the game scene.
         players.clear()
         server_disconnected.emit()
 
+ .. code-tab:: csharp C#
+    using Godot;
+    using Godot.Collections;
+
+    public partial class Lobby : Node
+    {
+        // These signals can be connected to by a UI lobby scene or the game scene.
+        [Signal]
+        public delegate void PlayerConnectedEventHandler(int peerId, Dictionary<string, string> playerInfo);
+
+        [Signal]
+        public delegate void PlayerDisconnectedEventHandler(int peerId);
+
+        [Signal]
+        public delegate void ServerDisconnectedEventHandler();
+
+        private const int _port = 7000;
+        private const string _defaultServerIp = "127.0.0.1";
+        private const int _maxConnections = 20;
+
+        // This will contain player info for every player, with the keys being each player's unique IDs.
+        private readonly Dictionary<long, Dictionary<string, string>> _players = new();
+
+        // This is the local player info. This should be modified locally before the connection is made.
+        // It will be passed to every other peer.
+        // For example, the value of "name" can be set to something the player entered in a UI scene.
+        private readonly Dictionary<string, string> _playerInfo = new()
+        {
+            {"name", "Name"}
+        };
+
+        private int _playersLoaded = 0;
+
+        public override void _Ready()
+        {
+            Multiplayer.PeerConnected += _OnPlayerConnected;
+            Multiplayer.PeerDisconnected += _OnPlayerDisconnected;
+            Multiplayer.ConnectedToServer += _OnConnectedOk;
+            Multiplayer.ConnectionFailed += _OnConnectedFail;
+            Multiplayer.ServerDisconnected += _OnServerDisconnected;
+        }
+
+        public int JoinGame(string address = "")
```

Comment:
Use the `Error` enum for the return type. Then there's no need to cast it to `int`.

```suggestion
        public Error JoinGame(string address = "")
```